.github/scripts/label_promoted_commits.py

```diff
@@ -23,70 +22,66 @@ def parser():
-            params = {
-                "q": query,
-            }
-            headers = {
-                "Authorization": f"token {github_token}",
-                "Accept": "application/vnd.github.v3+json"
-            }
-            response = requests.get(search_url, headers=headers, params=params)
-            prs = response.json().get("items", [])
-            for pr in prs:
-                refs_match = re.findall(r'Refs (?:#|https.*?)(\d+)', pr["body"])
-                ref = re.search(r'branch-(\d+\.\d+)', args.ref)
-                if refs_match and ref:
-                    pr_number = int(refs_match[0])
-                    backport_pr_number = int(match.group(1))
-                    if pr_number in processed_prs:
-                        continue
-                    label_to_add = f'backport/{ref.group(1)}-done'
-                    label_to_remove = f'backport/{ref.group(1)}'
-                    pr = repo.get_pull(pr_number)
-                    current_labels = pr.get_labels()
-                    current_label_names = [label.name for label in current_labels]
-                    if label_to_remove in current_label_names:
-                        pr.remove_from_labels(label_to_remove)
-                    if label_to_add not in current_label_names:
-                        pr.add_to_labels(label_to_add)
+            pr_number = int(match.group(1))
+            if pr_number in processed_prs:
+                continue
+            if target_branch:
+                pr = repo.get_pull(pr_number)
+                refs_pr = re.findall(r'Refs (?:#|https.*?)(\d+)', pr.body)
+                if refs_pr:
+                    print(f'branch-{target_branch.group(1)}, pr number is: {pr_number}')
+                    # 1. change the backport label of the parent PR to note that
+                    #    we've merge the corresponding backport PR
+                    # 2. close the backport PR and leave a comment on it to note
+                    #    that it has been merged with a certain git commit,
+                    ref_pr_number = refs_pr[0]
+                    mark_backport_done(repo, ref_pr_number, branch_name)
                     comment = f'Closed via {commit.sha}'
-                    add_comment_and_close_pr(backport_pr_number, repo, comment)
-                    processed_prs.add(pr_number)
-                else:
-                    assert re.match(r'/master', args.ref)
-                    pr_number = pr["number"]
-                    pr = repo.get_pull(pr_number)
-                    label_to_add = args.label
-                    current_labels = pr.get_labels()
-                    current_label_names = [label.name for label in current_labels]
-                    if label_to_add not in current_label_names:
-                        pr.add_to_labels(label_to_add)
-                processed_prs.add(pr_number)
+                    add_comment_and_close_pr(pr_number, repo, comment)
```

Comment:
we already have an instance of pull request with this `pr_number`, and it's named `pr`. can we pass it down to this function, so that we can save a call (http request).

Reply:
You are right, changed :-)